core/src/consensus.rs

```diff
@@ -211,6 +211,17 @@ impl TowerVersions {
     }
 }
 
+#[derive(PartialEq, Eq, Debug, Default, Clone, Copy, AbiExample)]
+pub(crate) enum BlockhashStatus {
+    /// No vote since restart
+    #[default]
+    Uninitialized,
+    /// Failed generation of last vote tx
+    Failed,
```

Comment:
Seems like this is just to cover the non-voter case, can we make this more specific and call it `NotVoting`